src/rules/no_dupe_args.rs

```diff
@@ -106,22 +97,20 @@ impl<'c, 'view> NoDupeArgsVisitor<'c, 'view> {
 
   fn check_params<'a, 'b, I>(&'a mut self, span: Span, params: I)
   where
-    I: Iterator<Item = &'b Param>,
+    I: Iterator<Item = &'b &'b Param<'b>>,
   {
     let pats = params.map(|param| &param.pat);
     self.check_pats(span, pats);
   }
 }
 
-impl<'ctx, 'view> VisitAll for NoDupeArgsVisitor<'ctx, 'view> {
-  noop_visit_type!();
-
-  fn visit_function(&mut self, function: &Function, _parent: &dyn Node) {
-    self.check_params(function.span, function.params.iter());
+impl Handler for NoDupeArgsHandler {
+  fn function(&mut self, function: &Function, _ctx: &mut Context) {
+    self.check_params(function.span(), function.params.iter());
```

Comment:
alternative is move map from check_params and delete check_params
```suggestion
    self.check_pats(function.span(), function.params.iter().map(|param| &param.pat));
```

Reply:
Either looks fine to me